modules/cli/src/main/scala/scala/cli/commands/Repl.scala

```diff
@@ -31,8 +31,34 @@ object Repl extends ScalaCommand[ReplOptions] {
   override def buildOptions(ops: ReplOptions): Some[BuildOptions] = Some {
     import ops.*
     import ops.sharedRepl.*
-    val baseOptions = shared.buildOptions().orExit(ops.shared.logger)
+
+    val ammoniteVersionOpt = ammoniteVersion.map(_.trim).filter(_.nonEmpty)
+    val baseOptions        = shared.buildOptions().orExit(ops.shared.logger)
     baseOptions.copy(
+      scalaOptions = baseOptions.scalaOptions.copy(
+        scalaVersion = baseOptions.scalaOptions.scalaVersion
+          .orElse {
+            val maxScalaVer     = scala.cli.internal.Constants.maxAmmoniteScala3Version
```

Comment:
if I'm not mistaken, this will only work for running `--amm` without specifying the Scala version explicitly.
it would be neat if we had the auto-downgrade kicking in when the user passes a version with `-S`, but without an explicit patch version, like:
```scala
scala-cli --amm -S 3
scala-cli --amm -S 3.2
scala-cli --amm -S 2.13
scala-cli --amm -S 2.12
```

Reply:
I started to think a bit about that, but this needs more work. I htink this can be done in a subsequent PR.